### PR TITLE
fix the kill all subprocess when trying to abort kubetest

### PR DESCRIPTION
**Pull request by cpanato**
fixes: https://github.com/kubernetes/test-infra/issues/1906

tested on a Mac, and was able to kill the process and not subprocess was running after that.

### CODE REVIEW

File: kubetest/process/process.go [hunk 1]
```diff
@@ -170,9 +170,16 @@ func (c *Control) FinishRunning(cmd *exec.Cmd) error {
 		select {
 		case <-sigChannel:
 			log.Printf("Killing %v(%v) after receiving signal", stepName, -cmd.Process.Pid)
-			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
+
+			pgid, err := syscall.Getpgid(cmd.Process.Pid)
+			if err != nil {
```
Comment (ixdy) [inline kubetest/process/process.go]: should this be `err == nil`?

**Comment by ixdy** [comment]
There are several places where we use `syscall.Kill` with `-cmd.Process.Pid`. should we refactor this to fix them all?

/ok-to-test

File: kubetest/process/process.go (hunk 1 above)
Comment (fejta) [reply kubetest/process/process.go]: When will this fail? Log the error?

File: kubetest/process/process.go [hunk 2]
```diff
@@ -170,9 +170,16 @@ func (c *Control) FinishRunning(cmd *exec.Cmd) error {
...
+				pgid = cmd.Process.Pid
+			}
+
+			if err := syscall.Kill(-pgid, syscall.SIGKILL); err != nil {
```
Comment (fejta) [inline kubetest/process/process.go]: Do we still want `-pgid` or should we instead just do `pgid = -cmd.Process.Pid`?

File: kubetest/process/process.go [hunk 3]
```diff
@@ -170,9 +170,16 @@ func (c *Control) FinishRunning(cmd *exec.Cmd) error {
 		select {
 		case <-sigChannel:
 			log.Printf("Killing %v(%v) after receiving signal", stepName, -cmd.Process.Pid)
-			if err := syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL); err != nil {
+
+			pgid, err := syscall.Getpgid(cmd.Process.Pid)
```
Comment (fejta) [inline kubetest/process/process.go]: Could you add a comment about why we need this?

File: kubetest/process/process.go (hunk 2 above)
Comment (cpanato) [reply kubetest/process/process.go]: We want to target the group id and not the particular process, and researching looks like this is the right thing to do.

File: kubetest/process/process.go (hunk 1 above)
Comment (cpanato) [reply kubetest/process/process.go]: maybe when the pid does not exist anymore? add a log to print the error if happens

File: kubetest/process/process.go (hunk 3 above)
Comment (cpanato) [reply kubetest/process/process.go]: done

**Pull request merged**